src/monitor/node_active_protocol.c

```diff
@@ -2178,7 +2178,7 @@ synchronous_standby_names(PG_FUNCTION_ARGS)
 
 		if (secondaryNode != NULL &&
 			secondaryNode->replicationQuorum &&
-			IsCurrentState(secondaryNode, REPLICATION_STATE_SECONDARY))
+			secondaryNode->goalState == REPLICATION_STATE_SECONDARY)
```

Comment:
Would this lead to any issues if we have any failures during `catching up -> secondary`. Though, in the worst case, the writes might be blocked until the failure is detected, which sounds OK to me. Do you see any other edge cases on such failure?

Reply:
I agree with your analysis, and failed to think of other edge cases. We already react properly to network splits and other failures by re-assigning CATCHING-UP to the failed or split secondary, so we should be fine. That said, thinking more about it, we might want to also change the test at the following place:

https://github.com/citusdata/pg_auto_failover/blob/master/src/monitor/group_state_machine.c#L813

Maybe that's better in another PR though, I'll merge this one and prepare a new one.

Reply:
See #493